modules/internal/DefaultRectangular.chpl

```diff
@@ -2055,9 +2057,13 @@ module DefaultRectangular {
     const Aidx = A.getDataIndex(Alo);
     const Adata = _ddata_shift(A.eltType, A.theData, Aidx);
     const Alocid = Adata.locale.id;
+    const Asublocid = if CHPL_LOCALE_MODEL != "gpu" then c_sublocid_any else
```

Comment:
I agree. But somehow I have a recollection where we were trying to avoid `_none` for some reason. I'll grep around a bit more, and may take your suggestion based on what I found.

Reply:
Soo, for a task, `_any` means the task didn't target any sublocale. `_none` means it specifically targeted the host. See:

```chpl
extern proc chpl_task_getRequestedSubloc(): int;

writeln(chpl_task_getRequestedSubloc());  // -2


const host = here;

on here.gpus[0] {
  writeln(chpl_task_getRequestedSubloc());  // 0

  on host {
    writeln(chpl_task_getRequestedSubloc());  // -1
  }
}

on here {
  writeln(chpl_task_getRequestedSubloc());  // -1
}
```

For the time being `_any` feels the right constant here as `_none` is more strict and I don't know the implications of that down the road. Today, though it shouldn't matter with this PR as the runtime is checking for `<0`.

I think we need to think a bit more about these constants at some point. Maybe renaming `_none` to `_cpu` is an improvement?

Reply:
To be clear, I don't intend to make that final suggestion happen in that PR. I like the idea but dunno the general implications on runtime implementation's clarity.

Reply:
> Soo, for a task, _any means the task didn't target any sublocale. _none means it specifically targeted the host. See:

This is my understanding too.

Right, so I guess this LOC is saying "if we're not using the gpu locale model don't make any assumptions about what sublocale (which may be, perhaps, NUMA domains) that this data is on.  Sounds right.

Anyway, you can consider this item resolved.